anyio/_backends/_trio.py

```diff
@@ -240,12 +246,19 @@ async def wait(self):
         await self._event.wait()
 
 
-class Condition(trio.Condition):
+class Condition:
+    def __init__(self, lock: trio.Lock = None):
```

Comment:
Use `Optional[trio.Lock]` for the type hint.

Reply:
I was lazy because AFAIK `Optional` is auto-implied when the default is `None`.

Reply:
https://docs.python.org/3/library/typing.html#typing.get_type_hints 

> Optional[t] is added for function and method annotations if a default value equal to None is set.

But I added it anyway because you're doing it everywhere else. ;-)